### PR TITLE
add generic analytics section for segment compatible api

### PR DESCRIPTION
**Pull request by the-newman**

### CODE REVIEW

File: mautrix_instagram/web/analytics.py
Comment (tulir) [inline mautrix_instagram/web/analytics.py:1]: What happened to segment.py 🤔 (or more specifically, why did nothing happen to it)?

File: mautrix_instagram/web/analytics.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    if analytics_token:
+        background_task.create(_track(user, event, properties or {}))
+
+
+def init(host: str | None, key: str | None, user_id: str | None = None):
+    log.debug("analytics is a go go")
+    if not host or not key:
+        return
+    global analytics_url, analytics_token, analytics_user_id, http
+    analytics_url = urlunparse(("https", host, "/v1/track", "", "", ""))
```
Comment (tulir) [inline mautrix_instagram/web/analytics.py:42]: Same yarl thing as facebook

File: mautrix_instagram/web/analytics.py
Comment (the-newman) [reply mautrix_instagram/web/analytics.py:1]: Nice spot! I thought I had deleted that file.

**Comment by the-newman** [comment]
Thanks for the review @tulir

**Pull request merged**